kratos/processes/apply_periodic_boundary_condition_process.h

```diff
@@ -0,0 +1,196 @@
+     */
+    void RemoveCommonNodesFromSlaveModelPart();
+
+    /**
+     * @brief   The function to figure out how the master and slave model parts relate together and add master-slave constraints
+     *          to the rModelPart
+     */
+    template <int TDim>
+    void ApplyConstraintsForPeriodicConditions();
+
+    /**
+     * @brief   The function add the linear master-slave constraint to rModelPart. This function is specifically for applying
+     *          periodic conditions for vector variable. This distinction is because, for vector variables the variable should also be
+     *          transformed according to the transfromation specified in the settings.
+     * @param rSalveNode The slave node which is to be connected to the rHostGeometry.
+     * @param rHostedGeometry the Host geometry which has the rSlaveNode.
+     * @param rWeights The weights with which the rSlaveNode is connected to the rHostedGeometry's nodes.
+     * @param rVarName The name of the vector variable on which periodic boundary condition can be applied.
+     */
+    template <int TDim>
+    void ConstraintSlaveNodeWithConditionForVectorVariable(NodeType& rSalveNode, const GeometryType& rHostedGeometry, const VectorType& rWeights, const std::string& rVarName );
+
+    /**
+     * @brief   The function add the linear master-slave constraint to rModelPart. This function is specifically for applying
+     *          periodic conditions for scalar variable. This distinction is because, for scalar variables the variable need NOT be
+     *          transformed according to the transfromation specified in the settings.
+     * @param rSalveNode The slave node which is to be connected to the rHostGeometry.
+     * @param rHostedGeometry the Host geometry which has the rSlaveNode.
+     * @param rWeights The weights with which the rSlaveNode is connected to the rHostedGeometry's nodes.
+     * @param rVarName The name of the scalar variable on which periodic boundary condition can be applied.
+     */
+    template <int TDim>
+    void ConstraintSlaveNodeWithConditionForScalarVariable(NodeType& rSalveNode, const GeometryType& rHostedGeometry, const VectorType& rWeights, const std::string& rVarName );
+
+    /**
+     * @brief   Fuctions calculates the transformation matrix to account for the moving the two periodic condition modelparts together.
+     */
+    void CalculateTransformationMatrix();
+
+    /**
+     * @brief   Calculates the transformation matrix which translates the given vector alone mDirOfTranslation by mModulus
+     * @param   Modulus is the magnitude by which the translation should happen in the direction of mDirOfTranslation.
+     * @param   rMatrix is the transformation matrix which will be calculated in this function. This should be of correct size (4x4).
+     */
+    void CalculateTranslationMatrix(const double Modulus, MatrixType& rMatrix);
```

Comment:
Isn't Modulus the same as mModulus? If yes, it should not be a parameter. Also if you packaged mModulus with the mDirofTranslation in a separate struct, then this function could be a member of the struct or you can make it a free function and pass the struct as a parameter. This will improve the code.